src/dialogs/quick-bar/ha-quick-bar.ts

```diff
@@ -181,16 +188,37 @@ export class QuickBar extends LitElement {
     }
 
     this._items = (this._commandMode ? this._commandItems : this._entityItems)
-      .filter(({ text, altText }) => {
-        const values = [text];
-        if (altText) {
-          values.push(altText);
-        }
-        return fuzzySequentialMatch(this._itemFilter, values);
-      })
-      .sort((itemA, itemB) => compare(itemA.text, itemB.text));
+      .filter(({ text, altText }) =>
+        altText
+          ? isPatternInWords(this._itemFilter, text, altText)
+          : isPatternInWords(this._itemFilter, text)
+      )
+      .map<ScoredItem>((item) => ({
+        item,
+        score: item.altText
+          ? fuzzySequentialMatch(this._itemFilter, item.text, item.altText)
+          : fuzzySequentialMatch(this._itemFilter, item.text),
+      }))
+      .sort(this.compareFuzzyMatches)
+      .map((scoredItem) => scoredItem.item);
```

Comment:
Should we make a helper for all this?

Reply:
We could, but it seemed like that refactor should happen in the next PR that wants to use the fuzzy filter.

Reply:
I don't think a helper will be necessary after recent push.

Reply:
Nevermind. I needed to put back the sorting logic, but I moved it into sequence matcher as a helper :)